iceoryx_posh/test/integrationtests/service_discovery/test_roudi_findservice.cpp

```diff
@@ -32,54 +32,154 @@ class RoudiFindService_test : public RouDiServiceDiscoveryTest
+}
+
+TEST_F(RoudiFindService_test, OfferServiceWithInvalidEventIdFails)
+{
+    auto isServiceOffered = senderRuntime->offerService(
+        iox::capro::ServiceDescription({"service1", "instance1", iox::capro::InvalidIDString}));
+    this->InterOpWait();
+
+    ASSERT_EQ(false, isServiceOffered);
+}
+
+TEST_F(RoudiFindService_test, OfferServiceWithAnyEventIdFails)
+{
+    auto isServiceOffered =
+        senderRuntime->offerService(iox::capro::ServiceDescription(123u, 456u, iox::capro::AnyEvent));
+    this->InterOpWait();
+
+    ASSERT_EQ(false, isServiceOffered);
+}
+
+TEST_F(RoudiFindService_test, OfferServiceWithAnyEventIdStringFails)
+{
+    auto isServiceOffered = senderRuntime->offerService(
+        iox::capro::ServiceDescription({"service1", "instance1", iox::capro::AnyEventString}));
+    this->InterOpWait();
+
+    ASSERT_EQ(false, isServiceOffered);
+}
+
+TEST_F(RoudiFindService_test, ReofferedServiceWithValidServiceDescriptionCanBeFound)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    senderRuntime->stopOfferService({"service1", "instance1"});
+    this->InterOpWait();
     senderRuntime->offerService({"service1", "instance1"});
     this->InterOpWait();
-    iox::runtime::InstanceContainer instanceContainer;
-    receiverRuntime->findService({"service1", "instance1"}, instanceContainer);
 
-    ASSERT_THAT(instanceContainer.size(), Eq(1u));
-    ASSERT_THAT(*instanceContainer.begin(), Eq(IdString_t("instance1")));
+    auto instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+
+    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));
+}
+
+TEST_F(RoudiFindService_test, OfferExsistingServiceMultipleTimesIsRedundant)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+
+    auto instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+
+    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));
+}
+
+TEST_F(RoudiFindService_test, FindSameServiceMultipleTimesReturnsSingleInstance)
+{
+    senderRuntime->offerService({"service1", "instance1"});
+    this->InterOpWait();
+
+    auto instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));
+
+    instanceContainer = receiverRuntime->findService({"service1", "instance1"});
+    ASSERT_THAT(instanceContainer.value().size(), Eq(1u));
+    ASSERT_THAT(*instanceContainer.value().begin(), Eq(IdString_t("instance1")));
 }
 
 TEST_F(RoudiFindService_test, DISABLED_OfferMultiMethodServiceSingleInstance_PERFORMANCETEST42)
```

Comment:
would it be possible to enable this test?

Reply:
enabled